toppingmaker/projecttopping.py

```diff
@@ -379,11 +382,80 @@ def make_items(
 
                 self[name] = maptheme_item
 
+    class Variables(dict):
+        """
+        A dict object of dict items describing a variable according to the variable keys listed in the ExportSettings passed on parsing the QGIS project.
+        """
+
+        def make_items(
+            self,
+            project: QgsProject,
+            export_settings: ExportSettings,
+        ):
+            self.clear()
+            for variable_key in export_settings.variables:
+                self[variable_key] = QgsExpressionContextUtils.projectScope(
+                    project
+                ).variable(variable_key)
+
+    class Layouts(dict):
+        """
+        A dict object of dict items describing a layout with templatefile according to the layout names listed in the ExportSettings passed on parsing the QGIS project.
+        """
+
+        class LayoutItemProperties(object):
+            """
+            The properties of a layout item.
+            Currently it's only a layout  template file. Maybe in future here as well items or whatever can be defined.
+            """
+
+            def __init__(self):
+                # the layout  template file - if None then not requested
+                self.templatefile = None
+
+        def __init__(self):
+            self.temporary_toppingfile_dir = os.path.expanduser("~/.temp_topping_files")
+
+        def make_items(
+            self,
+            project: QgsProject,
+            export_settings: ExportSettings,
+        ):
+            self.clear()
+
+            # go through all the print layouts in the project and export the requested ones
+            for layout in project.layoutManager().printLayouts():
+                if layout.name() in export_settings.layouts:
+                    self[layout.name()] = ProjectTopping.Layouts.LayoutItemProperties()
+
+                    filename_slug = f"{slugify(layout.name())}.qpt"
+                    os.makedirs(self.temporary_toppingfile_dir, exist_ok=True)
+                    temporary_toppingfile_path = os.path.join(
+                        self.temporary_toppingfile_dir, filename_slug
+                    )
+                    # clarify what needs to be set in the QgsReadWriteContext()
+                    layout.saveAsTemplate(
+                        temporary_toppingfile_path, QgsReadWriteContext()
```

Comment:
It could be interesting to log the contents of `context.takeMessages()`

Reply:
Do you have any docs about QgsReadWriteContext() to look up?

Of course the source code is always a way to go, but still...